lib/kubeclient/common.rb

```diff
@@ -78,11 +92,47 @@ def handle_exception
       raise KubeException.new(e.http_code, err_message, e.response)
     end
 
+    def discover
+      @entities = {}
+      response = handle_exception do
+        rest_client.get(@headers)
+      end
```

Comment:
That bugged me too.
It is done elsewhere so I submitted:
https://github.com/abonas/kubeclient/pull/186
(For the other one line handler)

and I will fix this one here